Assignment4/queries_SPARQL.txt

```diff
@@ -0,0 +1,119 @@
+Juan Gómez
+t110021
+
+NOTA: El prefijo común a todas las queries es: <http://GP-onto.fi.upm.es/exercise2#>
+prefix pr:<http://GP-onto.fi.upm.es/exercise2#>
+
+#1
+
+select distinct ?Concept where {[] a ?Concept} LIMIT 100
+
+#2
+
+select distinct ?x
+where
+{
+	?x rdfs:subClassOf pr:Establishment
+} LIMIT 100
+
+#3
+
+select distinct ?x
+where {
+	?x a pr:City
+} LIMIT 100
+
+#4
+
+select ?x
+where {
+pr:Santiago_de_Compostela pr:hasInhabitantNumber ?x
+} LIMIT 100
+
+#5
+
+select ?x ?y
+where {
+pr:Santiago_de_Compostela pr:hasInhabitantNumber ?x .
+pr:Arzua pr:hasInhabitantNumber ?y
+} LIMIT 100
+
+#6
+
+select distinct ?x ?y
+where {
+?x pr:hasInhabitantNumber ?y .
+} ORDER BY ASC(?x)
+LIMIT 100
+
+#7
+
+select distinct ?x ?y
+where {
+?z rdfs:subClassOf pr:Locality .
+?x rdfs:subClassOf ?z .
+OPTIONAL { ?x pr:hasInhabitantNumber ?y }
+} LIMIT 100
+
+#8
+
+select distinct ?x ?y
+where {
+?x rdfs:subClassOf pr:Locality .
+?x pr:hasInhabitantNumber ?y .
+FILTER (xsd:integer(?y) > xsd:integer("200000"))
+} LIMIT 100
+
```

Comment:
The second type cast of "200000" would not be needed if you just wrote 200000. In any case, the query provides correct results.